tracing-subscriber/src/registry/sharded.rs

```diff
@@ -0,0 +1,280 @@
+use sharded_slab::{Guard, Slab};
+
+use crate::{
+    fmt::span::SpanStack,
+    registry::{LookupSpan, SpanData},
+};
+use serde_json::{json, Value};
+use std::{
+    cell::RefCell,
+    collections::HashMap,
+    convert::TryInto,
+    fmt,
+    sync::{
+        atomic::{fence, AtomicUsize, Ordering},
+        Arc, Mutex,
+    },
+};
+use tracing_core::{
+    dispatcher,
+    field::{FieldSet, Visit},
+    span::{self, Id},
+    Event, Field, Interest, Metadata, Subscriber,
+};
+
+#[derive(Debug)]
+pub struct Registry {
+    spans: Arc<Slab<Data>>,
```

Comment:
I don't think the `Arc` is necessary. The registry will be a singleton that already lives inside an `Arc` (the dispatcher).

Reply:
That's a good point!